src/Core.Nhibernate/Extensions/IdentityServerServiceFactoryExtensions.cs

```diff
@@ -1,91 +1,65 @@
             }
         }
-        private static void RegisterOperationalServices(IdentityServerServiceFactory factory, NhibernateServiceOptions serviceOptions)
+        private static void RegisterOperationalServices(IdentityServerServiceFactory factory)
         {
-            if (factory == null) throw new ArgumentNullException(nameof(factory));
-            if (serviceOptions == null) throw new ArgumentNullException(nameof(serviceOptions));
-
             factory.AuthorizationCodeStore = new Registration<IAuthorizationCodeStore, AuthorizationCodeStore>();
             factory.TokenHandleStore = new Registration<ITokenHandleStore, TokenHandleStore>();
             factory.ConsentStore = new Registration<IConsentStore, ConsentStore>();
             factory.RefreshTokenStore = new Registration<IRefreshTokenStore, RefreshTokenStore>();
         }
 
-        private static void RegisterConfigurationServices(IdentityServerServiceFactory factory, NhibernateServiceOptions serviceOptions)
+        private static void RegisterConfigurationServices(IdentityServerServiceFactory factory)
         {
-            RegisterClientStore(factory, serviceOptions);
-            RegisterScopeStore(factory, serviceOptions);
-        }
-
-        private static void RegisterClientStore(IdentityServerServiceFactory factory, NhibernateServiceOptions serviceOptions)
-        {
-            if (factory == null) throw new ArgumentNullException(nameof(factory));
-            if (serviceOptions == null) throw new ArgumentNullException(nameof(serviceOptions));
-
-            RegisterSessionFactory(factory, serviceOptions);
-
             factory.ClientStore = new Registration<IClientStore, ClientStore>();
             factory.CorsPolicyService = new Registration<ICorsPolicyService, ClientConfigurationCorsPolicyService>();
-
-        }
-
-        private static void RegisterScopeStore(IdentityServerServiceFactory factory, NhibernateServiceOptions serviceOptions)
-        {
-            if (factory == null) throw new ArgumentNullException(nameof(factory));
-            if (serviceOptions == null) throw new ArgumentNullException(nameof(serviceOptions));
-
-            RegisterSessionFactory(factory, serviceOptions);
-
             factory.ScopeStore = new Registration<IScopeStore, ScopeStore>();
         }
 
-        private static void RegisterSessionFactory(IdentityServerServiceFactory factory, NhibernateServiceOptions serviceOptions)
+        private static void RegisterSessionFactory(
+            IdentityServerServiceFactory factory, 
+            ISessionFactory NhibernateSessionFactory,
+            IDbProfileConfig dbProfileConfig)
         {
             if (factory.Registrations.All(r => r.DependencyType != typeof(ISessionFactory)))
             {
-                factory.Register(
-                    new Registration<ISessionFactory>(serviceOptions.NhibernateSessionFactory));
+                factory.Register(new Registration<IDbProfileConfig>(dbProfileConfig));
```

Comment:
Should probably build the configuration here and and register the IMapper instead. Unless there's a more appropriate location.

Reply:
Have updated the approach.